### PR TITLE
For your consideration: pub only, custom topic path, and load constant payload from file

### PR DESCRIPTION
**Pull request by carlospsikick**
Hello:
I send this for you consideration. This changes have been very useful for our tests and maybe you can find some value in them too. There are 3 commits:

1.  setup a pub-only mode to skip the receiving part of the test cycle. I've been using this to help me pump data as fast as possible into my system. I wanted to disable the reception part because I'm measuring the performance of my own consumer against the broker, in isolation. 
2. set a custom topic path. This helps me activate my message routing during the load test. I just needed to set a base topic and let mqtt-stresser do the rest.
3. load the constant payload from a file. Following a convention I've seen in other tools, if the constant path string starts with "@" assume it's a file and load the payload from it. 

Like I said, these changes are useful in our usecase. It could be that this is not the best way to implement them (I'm open to ideas) but if you find any of this interesting let me know, I can send individual PRs if necessary.

I keep having fun with this tool, great work guys!

### CODE REVIEW

**Comment by hikhvar** [comment]
I really like the general idea of this PR. I think the generator only mode might be useful. I will take a look tomorrow.

File: main.go [hunk 1]
```diff
@@ -174,10 +177,25 @@ func main() {
...
 
+	if len(*argTopicBasePath) > 0 {
+		topicNameTemplate = strings.Replace(topicNameTemplate, "internal/mqtt-stresser", *argTopicBasePath, 1)
+	}
+
 	payloadGenerator := defaultPayloadGen()
 	if len(*argConstantPayload) > 0 {
-		verboseLogger.Printf("Set constant payload to %s\n", *argConstantPayload)
-		payloadGenerator = constantPayloadGenerator(*argConstantPayload)
+		if strings.Index(*argConstantPayload, "@") == 0 {
```
Comment (hikhvar) [inline main.go:186]: I think we should move this logic to the constantPayloadGenerator function or build a new function around it to keep things were they belong.

File: worker.go [hunk 2]
```diff
@@ -125,42 +128,44 @@ func (w *Worker) Run(ctx context.Context) {
 		setSkipTLS(subscriberOptions)
 	}
 
-	subscriber := mqtt.NewClient(subscriberOptions)
+	if w.PubSub {
```
Comment (hikhvar) [inline worker.go:131]: I think with all those big if/else scopes, we should start to move the subscribe and publish logic into different functions. This allows us to write something like this:
```
if w.PubSub {
   startSubscribe(....)
}

Comment (carlospsikick) [reply worker.go:131]: I like that approach. I'll work on it over the weekend.

File: main.go (hunk 1 above)
Comment (carlospsikick) [reply main.go:186]: good idea.

**Comment by carlospsikick** [comment]
Closing this PR to work on the proposed changes. I'll create a new one once I have something solid.
Thanks!

**Pull request closed without merging**